plat/renesas/rcar/aarch64/platform_common.c

```diff
@@ -0,0 +1,265 @@
+	MAP_ATFW_CRASH,
+	MAP_ATFW_LOG,
+	MAP_DEVICE_RCAR,
+	MAP_DEVICE_RCAR2,
+	MAP_SRAM,
+	MAP_SRAM_STACK,
+	{0}
+};
+#endif
+
+#if IMAGE_BL32
+const mmap_region_t rcar_mmap[] = {
+	MAP_DEVICE0,
+	MAP_DEVICE1,
+	{0}
+};
+#endif
+
+CASSERT(ARRAY_SIZE(rcar_mmap) + RCAR_BL_REGIONS
+	<= MAX_MMAP_REGIONS, assert_max_mmap_regions);
+
+/*
+ * Macro generating the code for the function setting up the pagetables as per
+ * the platform memory map & initialize the mmu, for the given exception level
+ */
+#if USE_COHERENT_MEM
+void rcar_configure_mmu_el3(unsigned long total_base,
+			    unsigned long total_size,
+			    unsigned long ro_start,
+			    unsigned long ro_limit,
+			    unsigned long coh_start,
+			    unsigned long coh_limit)
+{
+	mmap_add_region(total_base, total_base, total_size,
+			MT_MEMORY | MT_RW | MT_SECURE);
+	mmap_add_region(ro_start, ro_start, ro_limit - ro_start,
+			MT_MEMORY | MT_RO | MT_SECURE);
+	mmap_add_region(coh_start, coh_start, coh_limit - coh_start,
+			MT_DEVICE | MT_RW | MT_SECURE);
+	mmap_add(rcar_mmap);
+
+	init_xlat_tables();
+	enable_mmu_el3(0);
+}
+#else
+void rcar_configure_mmu_el3(unsigned long total_base,
+			    unsigned long total_size,
+			    unsigned long ro_start,
+			    unsigned long ro_limit)
+{
+	mmap_add_region(total_base, total_base, total_size,
+			MT_MEMORY | MT_RW | MT_SECURE);
+	mmap_add_region(ro_start, ro_start, ro_limit - ro_start,
+			MT_MEMORY | MT_RO | MT_SECURE);
+	mmap_add(rcar_mmap);
+
+	init_xlat_tables();
+	enable_mmu_el3(0);
+}
+#endif
+
+uintptr_t plat_get_ns_image_entrypoint(void)
+{
+#if (IMAGE_BL2)
+	uint32_t cert, len;
+	uintptr_t dst;
+	int32_t ret;
+
+	ret = rcar_get_certificate(NON_TRUSTED_FW_CONTENT_CERT_ID, &cert);
+	if (ret) {
+		ERROR("%s : cert file load error", __func__);
+		return NS_IMAGE_OFFSET;
+	}
+
+	rcar_read_certificate((uint64_t) cert, &len, &dst);
+
+	return dst;
+#else
+	return NS_IMAGE_OFFSET;
+#endif
+}
+
+unsigned int plat_get_syscnt_freq2(void)
+{
+	unsigned int freq;
+
+	freq = mmio_read_32(ARM_SYS_CNTCTL_BASE + CNTFID_OFF);
+	if (freq == 0)
+		panic();
+
+	return freq;
+}
+
+void plat_arm_gic_init(void)
+{
+	gicv2_distif_init();
+	gicv2_pcpu_distif_init();
+	gicv2_cpuif_enable();
+}
+
+static const unsigned int irq_sec_array[] = {
+#if IMAGE_BL2
+	ARM_IRQ_SEC_WDT		/* 173          */
+#else
+	ARM_IRQ_SEC_PHY_TIMER,	/* 29           */
+	ARM_IRQ_SEC_SGI_0,	/* 8            */
+	ARM_IRQ_SEC_SGI_1,	/* 9            */
+	ARM_IRQ_SEC_SGI_2,	/* 10           */
+	ARM_IRQ_SEC_SGI_3,	/* 11           */
+	ARM_IRQ_SEC_SGI_4,	/* 12           */
+	ARM_IRQ_SEC_SGI_5,	/* 13           */
+	ARM_IRQ_SEC_SGI_6,	/* 14           */
+	ARM_IRQ_SEC_SGI_7,	/* 15           */
+	ARM_IRQ_SEC_RPC,	/* 70           */
+	ARM_IRQ_SEC_TIMER,	/* 166          */
+	ARM_IRQ_SEC_TIMER_UP,	/* 171          */
+	ARM_IRQ_SEC_WDT,	/* 173          */
+	ARM_IRQ_SEC_CRYPT,	/* 102          */
+	ARM_IRQ_SEC_CRYPT_SecPKA,	/* 97           */
+	ARM_IRQ_SEC_CRYPT_PubPKA	/* 98           */
+#endif
+};
+
```

Comment:
You'll need to migrate this to the new arrays as done in https://github.com/ARM-software/arm-trusted-firmware/pull/1587